common/common_structs/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+#![no_std]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+pub type Nonce = u64;
+pub type Epoch = u64;
+
+#[derive(TopEncode, TopDecode, PartialEq, TypeAbi)]
+pub struct FftTokenAmountPair<BigUint: BigUintApi> {
+    pub token_id: TokenIdentifier,
+    pub amount: BigUint,
+}
+
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, PartialEq, TypeAbi, Clone)]
+pub struct GenericEsdtAmountPair<BigUint: BigUintApi> {
```

Comment:
Since it has `token_nonce` field, why call it GenericESDTAmountPair?

Reply:
NFTs and SFTs are ESDTs too. I'll rename the struct to GenericTokenAmountPair to avoid confusions

Reply:
ok. I thought ESDT is the ERC20 equivalent in Elrond. It is not the case?

Reply:
Fungible ESDT tokens are the equivalent of ERC20 since they have just a type (ID) and an amount. Non fungible ESDT tokens also have metadata and nonce and I think their equivalent would be ERC721, although I'm not 100% sure